core/base/mergeTreeClustering/MergeTreeDistance.h

```diff
@@ -0,0 +1,1504 @@
+      /*float m = 40;
+      float *nodesScalar1 = new float[8]{0*40, 1*40, 2*40, 3*40, 4*40, 5*40,
+      6*40, 7*40}; std::vector<SimplexId>      nodes1{0, 1, 2, 3, 4, 5, 6, 7};
+      std::vector<std::tuple<ftm::idNode, ftm::idNode>> arcs1{std::make_tuple(0,
+      6), std::make_tuple(5, 6), std::make_tuple(6, 7)};
+
+      float *nodesScalar2 = new float[8]{0*40, 2.5*40, 2.5*40,
+      3*40, 3.75*40, 5.75*40, 6*40, 7*40}; std::vector<SimplexId>      nodes2{0,
+      1, 2, 3, 4, 5, 6, 7}; std::vector<std::tuple<ftm::idNode, ftm::idNode>>
+      arcs2{std::make_tuple(0, 6), std::make_tuple(1, 5), std::make_tuple(4, 5),
+                                                              std::make_tuple(5,
+      6), std::make_tuple(6, 7)};
+
+      // barycenter
+      int mode = 2;
+      if(mode == 1){
+        //nodesScalar1 = new float[8]{0, 135, 135, 3*40, 170, 5.375*40, 6*40,
+      7*40}; nodesScalar1 = new float[8]{0, 150, 150, 3*40, 175, 5.375*40, 6*40,
+      7*40}; nodes1 = std::vector<SimplexId>{0, 1, 2, 3, 4, 5, 6, 7}; arcs1 =
+      std::vector<std::tuple<ftm::idNode, ftm::idNode>>{std::make_tuple(0, 6),
+                                                                  std::make_tuple(1,
+      5), std::make_tuple(4, 5), std::make_tuple(5, 6), std::make_tuple(6, 7)};
+      }
+      if(mode == 2){
+        //nodesScalar2 = new float[8]{0, 135, 135, 3*40, 170, 5.375*40, 6*40,
+      7*40}; nodesScalar2 = new float[8]{0, 150, 150, 3*40, 175, 5.375*40, 6*40,
+      7*40}; nodes2 = std::vector<SimplexId>{0, 1, 2, 3, 4, 5, 6, 7}; arcs2 =
+      std::vector<std::tuple<ftm::idNode, ftm::idNode>>{std::make_tuple(0, 6),
+                                                                  std::make_tuple(1,
+      5), std::make_tuple(4, 5), std::make_tuple(5, 6), std::make_tuple(6, 7)};
+      }*/
+
+      ///////////////////////////////////////////////////////////////////////////////
+
+      // Nodes of a pair map to different pairs 2
+      /*float *nodesScalar1 = new float[8]{0, 1, 2, 3, 4, 5, 6, 7};
+      std::vector<SimplexId>      nodes1{0, 1, 2, 3, 4, 5, 6, 7};
+      std::vector<std::tuple<ftm::idNode, ftm::idNode>> arcs1{std::make_tuple(0,
+      6), std::make_tuple(5, 6), std::make_tuple(6, 7)};
+
+      float *nodesScalar2 = new float[8]{0, 2, 3, 4, 3.75, 5.75, 6, 7};
+      std::vector<SimplexId>      nodes2{0, 1, 2, 3, 4, 5, 6, 7};
+      std::vector<std::tuple<ftm::idNode, ftm::idNode>> arcs2{std::make_tuple(0,
+      6), std::make_tuple(1, 3), std::make_tuple(2, 3), std::make_tuple(3, 5),
+                                                              std::make_tuple(4,
+      5), std::make_tuple(5, 6), std::make_tuple(6, 7)};*/
+
+      ///////////////////////////////////////////////////////////////////////////////
+
+      // Nodes of a pair map to different pairs 3
+      // COUNTER EXAMPLE edit distance
+      auto c = [&](int a) { return float(-a + 20 + 1); };
+      float *nodesScalar1 = new float[4]{0, c(20), c(13), 25};
```

Comment:
I believe this code can be commented right? (or even removed)
in general, please do **NOT** use `new` or `malloc` etc

Reply:
Indeed they are still some testing functions that can be removed. They are not used in the code, they were useful to me to test my code. Hence, `new` is only used here and nowhere else in the "real" code.